app.rb

```diff
@@ -209,6 +210,10 @@ def is_elasticsearch_available?
 end
 
 get '/heartbeat' do
+  Mongoid::Clients.clients.each do |name, client|
+    client.close
+    client.reconnect
+  end unless is_mongo_available?
```

Comment:
This will force a reconnect when mongodb is reported down or mongid is connected to a secondary node after a failover.